File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        n = len(prices)
        bestBuy = prices[0]
        maxProfit = 0
        for i in range(1,n):
            if prices[i] > bestBuy:
                maxProfit = max(maxProfit ,prices[i] - bestBuy)
            bestBuy = min(bestBuy , prices[i])       
        return maxProfit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minDay = prices[0];
        int maxProfit = 0;
        int n = prices.length;
        
        for(int i = 1; i < n; i++){
            minDay = Math.min(minDay,prices[i]);
            maxProfit = Math.max(maxProfit, prices[i] - minDay);
        }
        return maxProfit;
    }
}